src/Dict/Any.elm

```diff
@@ -631,3 +631,146 @@ encodeList encodeF =
 decodeList : (k -> comparable) -> Decode.Decoder ( k, v ) -> Decode.Decoder (AnyDict comparable k v)
 decodeList keyToComparable =
     Decode.map (fromList keyToComparable) << Decode.list
+
+
+{-| Takes a key-fn and a list.
+Creates an `AnyDict` which maps the key to a list of matching elements.
+-}
+groupBy : (value -> key) -> (key -> comparable) -> List value -> AnyDict comparable key (List value)
+groupBy toKey keyToComparable list =
+    List.foldr
+        (\x acc ->
+            update
+                (toKey x)
+                (\maybeValues ->
+                    maybeValues
+                        |> Maybe.map ((::) x)
+                        |> Maybe.withDefault [ x ]
+                        |> Just
+                )
+                acc
+        )
+        (empty keyToComparable)
+        list
+
+
+{-| Find out if there is any instance of something in a Dictionary.
+
+    type Animal = Cat | Mouse | Dog
+
+    animalToInt : Animal -> Int
+    animalToInt animal =
+        case animal of
+            Cat -> 0
+            Mouse -> 1
+            Dog -> 2
+
+    animals : AnyDict Int Animal String
+    animals =
+        [ (Cat, "Tom"), (Mouse, "Jerry") ]
+            |> fromList animalToInt
+
+    isACat : Animal -> String -> Bool
+    isACat animal _ =
+        case animal of
+            Cat -> True
+            _ -> False
+
+    any isACat animals
+    --> True
+
+-}
+any : (k -> v -> Bool) -> AnyDict comparable k v -> Bool
+any predicate dict =
+    foldl
```

Comment:
Might be worth it to use explicit recursion so that we can short circuit, in both `any` and `all`

Reply:
what about?
```elm
any predicate (AnyDict { dict }) =
    Dict.values dict |> List.any (uncurry predicate)
```
🤣

Reply:
I would also use explicit recursion. Also it would be great to get these functions to set-any as well to to keep some symmetry between the two.

Reply:
Like this?? 🤔 Is there a way to go entry by entry of the list without turning it into a list? 😵 
```elm
all : (k -> v -> Bool) -> AnyDict comparable k v -> Bool
all predicate (AnyDict { dict, toKey }) =
    case dict |> Dict.toList of
        [] ->
            True

        ( _, ( k, v ) ) :: rest ->
            predicate k v && all predicate (fromList toKey <| List.map Tuple.second rest)
```

Reply:
good point it seems there isn't. For that there would need to be some `uncons` or `pop` function which there isn't. So either we fold or we convert to list.

Reply:
Added a note regarding `Dict.Extra`, given that we would have to `Dict.toList` and `Dict.fromList` I think the current implementation with `foldl` is the way to go or the one using the internal dict [here](https://github.com/turboMaCk/any-dict/pull/16#discussion_r744016275) 😉

Reply:
agree